TLM/TLM/Manager/Impl/JunctionRestrictionsManager.cs

```diff
@@ -715,6 +717,102 @@ public class JunctionRestrictionsManager
+                if (otherSegmentId.ToSegment().Info.m_hasPedestrianLanes) {
+                    nearestClockwiseSegId = otherSegmentId;
+                    break;
+                }
+                ++clockwiseDistance;
+            }
+
+            if (nearestClockwiseSegId == 0) {
+
+                if (logLogic) {
+                    Log._Debug($"IsLikelyPedestrianRoute(segmentId: {segmentId}, startNode: {startNode}, includeAllPossible: {includeAllPossible})\r\t"
+                                + $"no pedestrian lanes on node {nodeId}"
+                                + $"returning false");
+                }
+
+                return false;
+            }
+
+            if (includeAllPossible) {
+
+                if (logLogic) {
+                    Log._Debug($"IsLikelyPedestrianRoute(segmentId: {segmentId}, startNode: {startNode}, includeAllPossible: {includeAllPossible})\r\t"
+                                + $"pedestrian lanes found on segment {nearestClockwiseSegId}"
+                                + $"returning true");
+                }
+
+                return true;
+            }
+
+            ushort nearestCounterClockwiseSegId = 0;
+            int counterClockwiseDistance = 0;
+            foreach (var otherSegmentId in ExtNodeManager.Instance.GetNodeSegmentIds(nodeId, ClockDirection.CounterClockwise, segmentId)) {
+                if (otherSegmentId.ToSegment().Info.m_hasPedestrianLanes) {
+                    nearestCounterClockwiseSegId = otherSegmentId;
+                    break;
+                }
+                ++counterClockwiseDistance;
+            }
+
+            if (nearestCounterClockwiseSegId == 0 || nearestClockwiseSegId == nearestCounterClockwiseSegId) {
+
+                if (logLogic) {
+                    Log._Debug($"IsLikelyPedestrianRoute(segmentId: {segmentId}, startNode: {startNode}, includeAllPossible: {includeAllPossible})\r\t"
+                                + $"no pedestrian lanes on segment {segmentId}"
+                                + $"only one segment on node {nodeId} has pedestrian lanes"
+                                + $"returning false");
+                }
+
+                return false;
+            }
+
+            int distanceThisSide = clockwiseDistance + counterClockwiseDistance - 1;
+            int distanceOtherSide = 0;
+            foreach (var otherSegmentId in ExtNodeManager.Instance.GetNodeSegmentIds(nodeId, ClockDirection.Clockwise, nearestClockwiseSegId)) {
+                ++distanceOtherSide;
+                if (otherSegmentId == nearestCounterClockwiseSegId)
+                    break;
+            }
+
+            bool result = distanceThisSide < distanceOtherSide;
+
+            if (logLogic) {
+                Log._Debug($"HasAnyPedestIsLikelyPedestrianRouterianAccess(segmentId: {segmentId}, startNode: {startNode}, includeAllPossible: {includeAllPossible})\r\t"
```

Comment:
wrong function name in the log.

Reply:
Yeah, I noticed that editing error. It will be fixed. 🙂